security/remote_devices/docker-swarm/docker-stack-edgex.yml

```diff
@@ -12,610 +10,991 @@
+    hostname: edgex-sys-mgmt-agent
+    image: nexus3.edgexfoundry.org:10004/sys-mgmt-agent:latest
+    networks:
+      edgex-network: {}
+    read_only: true
+    user: root:root
+    volumes:
+    - edgex-init:/edgex-init:ro,z
+    - edgex-secrets:/tmp/edgex/secrets
+    - /var/run/docker.sock:/var/run/docker.sock:z
+  ui:
     deploy: 
       placement: 
         constraints: 
-          - node.hostname == worker-node
-
-  device-rest:
-    image: edgexfoundry/docker-device-rest-go:1.1.1
-    ports:
-      - "49986:49986"
-    hostname: edgex-device-rest
-    networks:
-      edgex-network:
-        aliases:
-          - edgex-device-rest
+          - node.hostname != worker-node
     environment:
-      <<: *common-variables
-      Service_Host: edgex-device-rest
+      EDGEX_SECURITY_SECRET_STORE: "true"
+    hostname: edgex-ui-go
+    image: nexus3.edgexfoundry.org:10004/edgex-ui:latest
+    networks:
+      edgex-network: {}
+    ports:
+    - published: 4000
+      target: 4000
+      mode: host
+    read_only: true
+    user: 2002:2001
+  vault:
+    cap_add:
+    - IPC_LOCK
+    command: server
     depends_on:
-      - data
-      - command
+    - security-bootstrapper
     deploy: 
       placement: 
         constraints: 
-          - node.hostname == worker-node
+          - node.hostname != worker-node
+    entrypoint:
+    - /edgex-init/vault_wait_install.sh
+    environment:
+      API_GATEWAY_HOST: edgex-kong
+      API_GATEWAY_STATUS_PORT: '8100'
+      PROXY_SETUP_HOST: edgex-security-proxy-setup
+      STAGEGATE_BOOTSTRAPPER_HOST: edgex-security-bootstrapper
+      STAGEGATE_BOOTSTRAPPER_STARTPORT: '54321'
+      STAGEGATE_DATABASE_HOST: edgex-redis
+      STAGEGATE_DATABASE_PORT: '6379'
+      STAGEGATE_DATABASE_READYPORT: '6379'
+      STAGEGATE_KONGDB_HOST: edgex-kong-db
+      STAGEGATE_KONGDB_PORT: '5432'
+      STAGEGATE_KONGDB_READYPORT: '54325'
+      STAGEGATE_READY_TORUNPORT: '54329'
+      STAGEGATE_REGISTRY_HOST: edgex-core-consul
+      STAGEGATE_REGISTRY_PORT: '8500'
+      STAGEGATE_REGISTRY_READYPORT: '54324'
+      STAGEGATE_SECRETSTORESETUP_HOST: edgex-security-secretstore-setup
+      STAGEGATE_SECRETSTORESETUP_TOKENS_READYPORT: '54322'
+      STAGEGATE_WAITFOR_TIMEOUT: 60s
+      VAULT_ADDR: http://edgex-vault:8200
+      VAULT_CONFIG_DIR: /vault/config
+      VAULT_UI: "true"
+      SKIP_SETCAP: "true"
+      VAULT_DISABLE_MLOCK: "true"
+    hostname: edgex-vault
+    image: vault:1.8.5
+    networks:
+      edgex-network: {}
+    user: root:root
+    volumes:
+    - edgex-init:/edgex-init:ro,z
+    - vault-file:/vault/file:z
+    - vault-logs:/vault/logs:z
+version: '3.7'
+volumes:
+  consul-acl-token: {}
+  consul-config: {}
+  consul-data: {}
+  db-data: {}
+  edgex-init: {}
+  kong: {}
+  kuiper-connections: {}
+  kuiper-data: {}
+  kuiper-sources: {}
+  postgres-config: {}
+  postgres-data: {}
+  redis-config: {}
+  vault-config: {}
+  vault-file: {}
+  vault-logs: {}
+  edgex-secrets: {}
```

Comment:
Should we be using this approach in our docker compose deployments rather than local /tmp/...?